src/mon/MonClient.cc

```diff
@@ -429,10 +429,10 @@ void MonClient::handle_monmap(MMonMap *m)
   auto p = m->monmapbl.cbegin();
   decode(monmap, p);
```

Comment:
This probably is not that significant but decoding the monmap when we could know the epoch has not changed is wasteful. You could add an epoch to the `MMonMap` message to avoid decoding if it has not changed.

Reply:
Note, this also applies to `handle_quorum_peon`.

Reply:
will also add the epoch to the message so we won't need to re decode already known monmap